services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/SegmentServiceImpl.java

```diff
@@ -104,43 +107,46 @@ public Long addVlanEntity(String vlanId, String networkType, String vpcId) throw
      */
     @Override
     @DurationStatistics
-    public Long addVxlanEntity(String vxlanId, String networkType, String vpcId) throws Exception {
+    public Long addVxlanEntity(String vxlanId, String networkType, String vpcId, Integer mtu) throws Exception {
 
         Long key = null;
-        String rangeId = null;
+        String partitionStringFormat = null;
         Random ran = new Random();
-        Map<Integer, String> partitionsAndRangeIds = new HashMap<>();
+//        Map<Integer, String> partitionsAndRangeIds = new HashMap<>();
 
         try {
             NetworkVxlanType vxlan = new NetworkVxlanType();
+            // the number of randomly allocating partition in vxlan range
+            int round = 1;
+
+            while (round <= ConstantsConfig.MAX_ROUNDS) {
+                // Randomly allocate a partition and check if the partition exist in db
+                int partition = ran.nextInt(NetworkType.VXLAN_PARTITION);
```

Comment:
will the value of "partition" get changed for every round?

Also please use String method to convert to a String object.

Reply:
Yes，I could use String.valueOf()